### ISSUE TITLE
Support for asserting that an `exec_mock` was called

### ISSUE THREAD

**Issue by gruyaume** [canonical/ops-scenario#180]
## Description

I am using scenario's `exec_mock` and I can't figure out how to check that my mock was called.  Reading the documentation it is not clear whether this is possible. 

```python
class TestCharmConfigure(RouterUnitTestFixtures):
    def test_given_can_connect_when_configure_then_ip_forwarding_is_set(
        self,
    ):
        ip_forward_call = ("sysctl", "-w", "net.ipv4.ip_forward=1")
        container = scenario.Container(
            name="router",
            can_connect=True,
            exec_mock={
                ip_forward_call: scenario.ExecOutput(
                    return_code=0, stdout="net.ipv4.ip_forward = 1"
                ),
            },
        )
        state_in = scenario.State(
            leader=True,
            containers=[container],
        )

        state_out = self.ctx.run(container.pebble_ready_event, state_in)

        state_out.containers[0].exec_mock[ip_forward_call].assert_called()
```

## Logs
```
AttributeError: 'ExecOutput' object has no attribute 'assert_called'
```

## Reference
- https://github.com/canonical/ops-scenario

**Comment by tonyandrewmeyer** [comment]
I don't think this is possible in 6.x (without using a regular unittest/pytest mock).

However, you can do this in (very soon to be released) 7.0:

```python
class TestCharmConfigure(RouterUnitTestFixtures):
    def test_given_can_connect_when_configure_then_ip_forwarding_is_set(
        self,
    ):
        ip_forward_call = ("sysctl", "-w", "net.ipv4.ip_forward=1")
        container = scenario.Container(
            name="router",
            can_connect=True,
            execs={
                scenario.Exec(
                    ip_forward_call, return_code=0, stdout="net.ipv4.ip_forward = 1"
                ),
            },
        )
        state_in = scenario.State(
            leader=True,
            containers={container},
        )

        state_out = self.ctx.run(self.ctx.on.pebble_ready(container), state_in)

        assert self.ctx.exec_history[container.name][0].command == ip_forward_call
```

**Comment by tonyandrewmeyer** [comment]
7's out now, so this should be fixed.
